app/decorators/manageiq/providers/infra_manager_decorator.rb

```diff
@@ -0,0 +1,18 @@
+module ManageIQ::Providers
+  class InfraManagerDecorator < ExtManagementSystemDecorator
+    def quadicon(_n = nil)
+      {
+        :top_left     => { :text => hosts.size },
+        :top_right    => { :text => total_vms },
```

Comment:
I don't see the original code, so I am wondering why hosts.size is called, and then total_vms (without the call to size method) ? does it work in openstack provider too where there are instances? or they are also called vms in miq?

Reply:
Well, this comes from [app/helpers/quadicon_helper.rb:570](https://github.com/ManageIQ/manageiq-ui-classic/pull/3046/files#diff-6525cb9600835fcc2ba062fde39a2292L570) and `EmsInfra` is alias to [ManageIQ::Providers::InfraManager](https://github.com/ManageIQ/manageiq/blob/master/app/models/aliases/ems_infra.rb)